arch/arm/include/cortex_a_r/kernel_arch_func.h

```diff
@@ -50,12 +50,30 @@ static ALWAYS_INLINE void arch_kernel_init(void)
 #endif /* CONFIG_ARM_AARCH32_MMU */
 }
 
+#ifndef CONFIG_USE_SWITCH
+
 static ALWAYS_INLINE void
 arch_thread_return_value_set(struct k_thread *thread, unsigned int value)
 {
 	thread->arch.swap_return_value = value;
 }
 
+#else
+
+static inline void arch_switch(void *switch_to, void **switched_from)
```

Comment:
You want ALWAYS_INLINE here.  This gets expanded in just a few hot paths in the kernel; we know we want it inlined, don't give the toolchain the option to mess it up.

Reply:
Done